test/functional/legacy/096_location_list_spec.lua

```diff
@@ -0,0 +1,191 @@
+-- Test for problems in quickfix/location list:
+-- A. incorrectly copying location lists which caused the location list to show a
+--    different name than the file that was actually being displayed.
+-- B. not reusing the window for which the location list window is opened but
+--    instead creating new windows.
+-- C. make sure that the location list window is not reused instead of the window
+--    it belongs to.
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+describe('location list', function()
+  setup(clear)
+
+  it('is working', function()
+    -- Set up the test environment.
+    source(
+      -- This function serves as a callback which is executed on editing a new
+      -- buffer. It accepts a "test protocol" file name that looks like
+      -- "test://foo.txt". It sets some buffer-local settings and populates the
+      -- buffer with one line consisting of the base name ("foo").
+      [[
+      function! ReadTestProtocol(name)
+        let base = substitute(a:name, '\v^test://(.*)%(\.[^.]+)?', '\1', '')
+        let word = substitute(base, '\v(.*)\..*', '\1', '')
+      
+        setl modifiable
+        setl noreadonly
+        setl noswapfile
+        setl bufhidden=delete
+        %del _
+        " For problem 2:
+        " 'buftype' has to be set to reproduce the constant opening of new windows.
+        setl buftype=nofile
+      
+        call setline(1, word)
+      
+        setl nomodified
+        setl nomodifiable
+        setl readonly
+        exe 'doautocmd BufRead ' . substitute(a:name, '\v^test://(.*)', '\1', '')
+      endfunction
+      ]] ..
+
+      -- Register the above buffer setup function to be executed before
+      -- starting to edit a new "test protocol" buffer.
+      [[
+      augroup testgroup
+        au!
+        autocmd BufReadCmd test://* call ReadTestProtocol(expand("<amatch>"))
+      augroup END
+      ]] ..
+
+      -- Populate the location list of the current window with some test
+      -- protocol file locations such as "test://foo.txt", but without setting
+      -- line number locations.
+      [[
+      let words = [ "foo", "bar", "baz", "quux", "shmoo", "spam", "eggs" ]
+      let qflist = []
+      for word in words
+        call add(qflist, {'filename': 'test://' . word . '.txt', 'text': 'file ' . word . '.txt', })
+        " NOTE: problem 1:
+        " Intentionally not setting 'lnum' so that the quickfix entries are not
+        " valid.
+        call setloclist(0, qflist, ' ')
+      endfor
+    ]])
+
+    -- Set up the result buffer "test.out". This differs from the legacy test:
+    -- in order to avoid writing to disk, use ":file test.out" instead of
+    -- ":w! test.out" and adjust 'bufhidden' to avoid E37.
+    execute('enew')
+    execute('file test.out')
+    execute('setlocal bufhidden=hide')
```

Comment:
> In order to remove the test.out file on disk I had to change the bufhidden setting for the buffer.

Since you already did it, it's fine for this case. I am not sure if this should be a standard practice though... although it should be fine. There's nothing inherently better about writing it to disk.

> Ok if I leave it as one it?

Yes